docs/en/02_Developer_Guides/01_Templates/11_Partial_Template_Caching.md

```diff
@@ -0,0 +1,282 @@
+---
+title: Partial Template Caching
+summary: Cache a section of a template Reduce rendering time with cached templates and understand the limitations of the ViewableData object caching.
+icon: tags
+---
+
+## Partial template caching
+
+Partial template caching is a feature that allows caching of rendered portions of templates. Cached content
+is fetched from a [cache backend](../performance/caching), instead of being regenerated repeatedly.
+
+
+### Base syntax
+
+Here is an example of the basic usage. This is not a definitive example of the syntax, but it shows the most
+common use case.
+
+[note]
+See also [Complete Syntax definition](#complete-syntax-defintition) section
+[/note]
+
+```ss
+<% cached $CacheKey if $Condition %>
+  <!-- Content block -->
+<% end_cached %>
+```
+
+The key parts are `$CacheKey`, `$Condition` and `<!-- Content block -->`.  
+The following sections explain every one of them in more detail.
+
+
+#### $CacheKey
+
+The `$CacheKey` is a string identifying the block content within the cache storage.
+
+The form is `[$key[, $...]]`, where `$key` is any valid template expression and you can have any number of them delimited by comma.
+
+Ultimate value of the cache key includes the following items:
+  - hash of `SilverStripe\View\SSViewer::$global_key` rendered within current template context
+  - hash of unrendered `<!-- Content block -->`
+  - hash of concatenated `$CacheKey` elements
+
+`$CacheKey` is optional.  
+
+[note]
+Even if `$CacheKey` is omitted, hash of `<!-- Content block -->` and `SilverStripe\View\SSViewer::$global_key` values are still
+getting used to generate cache key for the caching backend storage.  
+[/note]
+
+[note]
+Using of a hash of `<!-- Content block -->` allows to invalidate cache when template changes.
+[/note]
+
+[warning]
+Avoid heavy computations in `$CacheKey` as it is evaluated for every template render.
+[/warning]
+
+
+#### $Condition
+
+The `$Condition` value is a boolean flag controlling if the block needs to cache.
+
+If the value is `false`, the block skips `$CacheKey` evaluation, does not lookup the data in the cache storage,
+neither preserve any data in the storage. The template within the block keeps working as is, same as without
+`<% cached %>` block surrounding it.
+
+The `$Condition` is optional (if omitted, `true` is implied).
+
+[note]
+Although `$Condition` is optional, it is strongly recommended.  
+Some cache backends do not cleanup expired cache records (e.g. the default file storage).
+If you rely on `$CacheKey` uniqueness rather than `$Condition`, you may
+exceed storage capacity if you do not clean up the outdated records manually.
```

Comment:
How are you supposed to manually clean up cached records on managed hosting?

Reply:
We cannot give any such recommendations in this documentation. That depends on too many factors specific for every application and its environment. We should still point that out, as this is a problem users should be aware of (we have precedents with bad consequences).
In the first place, our recommendation of using `$Condition` wherever possible should mitigate this problem to some degree. Further than that, changing the cache backend or clean-up of the filesystem is up to the user.

Reply:
> changing the cache backend or clean-up of the filesystem is up to the user.

Not for Silverstripe Ltd customers on managed hosting it's not =)  Diskspace management is more of an ops thing

Still I guess it's good to point and and recommend only cache when you need to.  I guess just remove the bit up '[you] clean up manually' since a lot of the people reading this won't actually be able to do that

just change it to disk space may run out?

Reply:
> Not for Silverstripe Ltd customers on managed hosting it's not 

That's out of context for this documentation. Here we're talking about the open-source project that may be used in many different environments.

>  I guess just remove the bit up '[you] clean up manually' since a lot of the people reading this won't actually be able to do that

yeah, fair enough. I'll think how to rewrite this whole note. This is only relevant for the "default" cache backend, but cleaning up is important even if you cache sparingly. I reckon we should keep it, but explain better